### PR TITLE
USHIFT-1459: Add cncf periodic job

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests

File: ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml [hunk 1]
```diff
@@ -1,4 +1,77 @@
...
+  - base_ref: main
+    org: openshift
+    repo: microshift
+  labels:
+    ci-operator.openshift.io/cloud: aws
+    ci-operator.openshift.io/cloud-cluster-profile: aws
+    ci-operator.openshift.io/cluster: build02
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-microshift-main-metal-tests-weekly
```
Comment (dhellmann) [inline ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml:18]: We need this to run daily.

Comment (pacevedom) [reply ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml:18]: Done.

File: ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh [hunk 2]
```diff
@@ -31,5 +31,10 @@ scp "${SETTINGS_FILE}" "${INSTANCE_PREFIX}:/home/${HOST_USER}/microshift/test/"
...
 trap 'scp -r ${INSTANCE_PREFIX}:/home/${HOST_USER}/microshift/_output/test-images/scenario-info ${ARTIFACT_DIR}' EXIT
 
+MODE="presubmits"
+if [[ "$JOB_NAME" == periodic* ]]; then
+  MODE="periodics"
+fi
+
 # Run the in-repo ci phase script to create the VMs for the test scenarios.
-ssh "${INSTANCE_PREFIX}" "/home/${HOST_USER}/microshift/test/bin/ci_phase_iso_boot.sh"
+ssh "${INSTANCE_PREFIX}" "MODE=${MODE} /home/${HOST_USER}/microshift/test/bin/ci_phase_iso_boot.sh"
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh]: As discussed, following the current customization approach of overriding almost anything through env vars, and the fact that this is common and it needs to run from 2 different scripts.

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests

**Comment by pacevedom** [comment]
/pj-rehearse periodic-ci-openshift-microshift-main-metal-tests-nightly pull-ci-openshift-microshift-main-microshift-metal-tests

**Comment by dhellmann** [comment]
This just adds the periodic variant of the existing job, right? I don't see CNCF results in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41823/rehearse-41823-periodic-ci-openshift-microshift-main-metal-tests-nightly/1687141792882888704/artifacts/metal-tests-nightly/openshift-microshift-e2e-metal-tests/artifacts/scenario-info/ but I think that's OK if we can test the periodic scenario PR with this new job somehow.

**Comment by pacevedom** [comment]
> This just adds the periodic variant of the existing job, right? I don't see CNCF results in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41823/rehearse-41823-periodic-ci-openshift-microshift-main-metal-tests-nightly/1687141792882888704/artifacts/metal-tests-nightly/openshift-microshift-e2e-metal-tests/artifacts/scenario-info/ but I think that's OK if we can test the periodic scenario PR with this new job somehow.

It is the existing job but it selects different test scenarios based on the job name (whether its periodic or not).
I found out I didnt consider rehearsal jobs having a prefix before the "periodic" keyword I was looking for. Will rehearse again with this change and you should see the results. I executed this manually and got the junit and everything.

**Comment by pacevedom** [comment]
/pj-rehearse periodic-ci-openshift-microshift-main-metal-tests-nightly

**Comment by pacevedom** [comment]
Job worked, here is the junit for cncf conformance: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/41823/rehearse-41823-periodic-ci-openshift-microshift-main-metal-tests-nightly/1687225135687274496

/cc @ggiguash @pmtk @dhellmann

**Comment by pmtk** [comment]
I'm wondering if we want a presubmit for periodics if there's a change in `test/scenarios-periodics/`? Otherwise the feedback loop is "no feedback on PR' -> wait for nightly

**Pull request merged**